modules/gateway/synchronize.go

```diff
@@ -83,14 +82,21 @@ func (g *Gateway) sendBlocks(conn modules.NetConn) (err error) {
 			break
 		}
 	}
-	if !found {
-		// The genesis block should be included in knownBlocks - if no matching
-		// blocks are found, the caller is probably on a different blockchain
-		// altogether.
-		return errors.New("no matching block found")
+	// If we didn't find any matching blocks, or if we're already
+	// synchronized, don't send any blocks. The genesis block should be
+	// included in knownBlocks, so if no matching blocks are found, the caller
+	// is probably on a different blockchain altogether.
```

Comment:
yes, so `found` will be true, but `start` will be greater than `g.state.Height()` (actually it should be exactly `g.state.Height() + 1`, but no harm in being safe).